src/vehicle.cpp

```diff
@@ -5839,12 +5839,15 @@ void vehicle::calc_mass_center( bool use_precalc ) const
         }
 
         units::mass m_part = 0_gram;
+        units::mass m_part_items = 0_gram;
         m_part += vp.part().base.weight();
         for( const auto &j : get_items( i ) ) {
-            //m_part += j.type->weight;
-            // Change back to the above if it runs too slowly
-            m_part += j.weight();
+            m_part_items += j.weight();
         }
+        if( vp.part().info().cargo_modifier != 100 ) {
+            m_part_items *= ( vp.part().info().cargo_modifier / 100 );
```

Comment:
```suggestion
            m_part_items *= vp.part().info().cargo_modifier / 100;
```
And you don't even need the `if` clause around it, just do the calculation unconditionally. If `cargo_modifier` is 100 it will become `m_part_items *= 1`, which is effectively not changing the value at all.

Reply:
I did "if" because I don't trust floats, probably because doubles, so I wanted to avoid changing the weight if 100/100 would not be 1 exactly.
Should also be a bit faster since mostly the modifier will be 100.